swim/state_transitions.go

```diff
@@ -46,30 +47,40 @@ type transitionTimer struct {
 type stateTransitions struct {
 	sync.Mutex
 
-	node *Node
-
+	node     *Node
+	clock    clock.Clock
```

Comment:
I don't think that's a good idea ([LoD](https://en.wikipedia.org/wiki/Law_of_Demeter)), even if I see that pattern used virtually everywhere in the codebase. It makes it hard to mock/test things and adds unnecessary coupling.